Code/GraphMol/AddHs.cpp

```diff
@@ -1078,6 +1078,22 @@ enum class HydrogenType {
   QueryHydrogen
 };
 
+template<class Q>
+std::pair<bool,bool> queryHasHs(Q queryAtom, bool inor=false) {
+  for(auto childit = queryAtom->beginChildren(); childit != queryAtom->endChildren(); ++childit) {
+    QueryAtom::QUERYATOM_QUERY::CHILD_TYPE query = *childit;
+    if (query->getDescription() == "AtomOr") {
+      return queryHasHs(query, true);
+    } else if (query->getDescription() == "AtomAtomicNum") {
```

Comment:
It can't currently happen when coming from SMARTS, but I think you should also cover `"AtomType"` here

Reply:
@greglandrum I'm coming back to this one, it fell off my queue.

Reply:
After some digging, you are correct, I think you've been doing the rdkit for too long :)

```
sma = "[#6]-[c,s,p]";
AtomAtomicNum 6 = val

AtomOr
  AtomOr
    AtomType 1006 = val
    AtomType 1016 = val
  AtomType 1015 = val
```